### ISSUE TITLE
NLO-to-NLO reweighting

### ISSUE THREAD

**Issue by ktht** [HEP-KBFI/hh-multilepton#40]
The idea is to treat NLO HH samples the same way as we treat LO HH samples. Ideally, we'd want to run the analysis such that NLO samples are both summed and kept separate (for validating 1D kl-scan, for instance). This entails the following:
1. duplicate the entries of NLO samples in the sample dictionaries with `reclassifySamples.py`
    - group samples with the same HH decay mode together, and give them a new unique sample category and name
    - might also want to add an additional field that tells whether or not we want to rewight the samples (can be useful in point 2)
    - ~set~ normalize the cross sections of the duplicated NLO samples to 1 pb -- just like we do for the LO samples
2. modify `is_nonresonant` function that tells the python FW to enable or disable HH reweighting
3. recreate new NLO denominators (= with coarser binning) in order to include 2D histograms for the NLO samples
4. (optional) improve `testHHWeightInterfaceNLO` to include an example of NLO-to-NLO reweighting

This is all a solid 1-2 days of work with testing included, as we need to make sure that the results we currently have remain unchanged.

**Comment by ktht** [comment]
Done; example (note the `nlo` in directory names):
![Screenshot_20210714_165540](https://user-images.githubusercontent.com/6233872/125634541-5da3666e-de24-446b-bbfe-0ec452da2bef.png)

**Comment by ktht** [comment]
Reopening because the current denominator histograms are created by counting the events in each (mHH, cosThetaStar) bin, not taking the gen weights into account. The NLO HH samples have quite broad distribution of gen weights, many with at least one or two outliers (see [here](https://github.com/HEP-KBFI/tth-htt/issues/158#issuecomment-706768579)). Thus, the way we handle gen weights at the analysis level has to be refactored.

**Comment by ktht** [comment]
Ok, I got confused for a moment. There is no need to refactor anything: the event weights don't play a role in the reweighting procedure, except in creating the denominator histograms. (Although, there are places in our code where we use unclipped gen weights such as in filling a histogram of gen weights or filling a branch of an Ntuple that is used as input to BDT training. If there's an outlier, then in the former case overflow bins are filled while in the latter case the outliers should be taken care of during the training.)

The real problem was that the differential NLO cross section was computed using the number of entries in the denominator histogram while it should be using the sum of all bin contents of the histogram. The two are different if there event weights are not equal to 1, as is the case with NLO samples.

This is now taken care of and all denominator histograms with coarser binning have been updated accordingly both in this repository and in bbWW repository. Tested 3l HH analysis on 2016 samples and found that before fixing the issue, the normalization of NLO signal was 22% smaller compared to LO signal, while now it's 7% smaller. So, there's some improvement.
